### DISCUSSION TITLE
Step 2) Udtræk-revision

### DISCUSSION THREAD

**Discussion by ReneTandsbjerg** [Kortforsyningen/FIRE#329] [Ideas]
OBS: Nedenfor nævnte punkter skal ses som en plan B i ændringer vi ønsker foretaget.
Plan A, som Palle og jeg ser det, vil være at kunne indlæse datasættet.xlsx fra revisions udtrækket ind i QGIS (liggende lokalt), for der igennem at rette og tilføje i QGIS som så ændre direkte ned i excel-arket og efter endt revision upload og indsætte via "fire niv ilæg revision projektnavn bxxxxxx"
Om det lader sig gøre ved jeg ikke, men det vil være rigtig smart hvis det kunne måske der er nogle "hajer" herinde der måske kunne bidrage med lidt viden om emnet.
 
1)Når vi laver et udtræk til revision kan det så laves sådan i regnearket at x´et i kolonne H(ikke besøgt) bliver fjernet. 
denne ændring ønskes fordi Revisions året samt uge og initialer er indikator for at punktet er besøgt.
x funktionen bliver relevant der hvor et punkt ikke besøges eksempelvis en kalot som ligger så langt væk fra de øvrige punkter vi revidere at det ikke vil give mening og derfor vil i beskrivelsen vil blive tilføjet "punkt vedligeholdes ikke" samt et x i kolonne H (ikke besøgt).
2) Beskrivelsen i kolonne D skal også stå i kolonne F (Ny værdi) og teksten rettes til i F (eksempelvis Adressen.)
3)Højde_over_terræn skal foruden C også stå i F og rettes til i F.
4)AFM: i kolonne D (eks: Plade mærket G.S.) skal også stå i kolonne F.
5)ATTR:bemærkning i kolonne D (eks: Rev. uge 17 2020 RTA.) skal også stå i kolonne F
Derefter vil det i revision-øjemed være sådan at vi kun skal slette i beskrivelsen i F hvis der ikke skal ændres.
tilføje et x i H ved de få punkter som ikke bliver besøgt af den ene eller anden årsag.
slette AFM: i F hvis punkttype er korrekt.
og i F ændre uge år og initialer udfor rækken ATTR: bemærkning.

Måske skal "uge" helt udelades fremadrettet, ser ikke den har nogen værdi som sådan.

Kom frisk
Vh René

**Comment by ReneTandsbjerg** [comment]
OBS: Nedenfor nævnte punkter skal ses som en plan B i ændringer vi ønsker foretaget såfremt PLAN A ikke kan realiseres

**Comment by kbevers** [comment]
En løsning med QGIS bliver ikke lavet lige foreløbigt. QGIS kan ikke rigtigt åbne et excelark så der skal opfindes en helt anden måde at håndtere data på end den nuværende hvis jeres ønske skal opfyldes. Det er muligt vi kommer der til i FIRE version 2, men lige nu bliver det ikke.

Jeg tror jeg forstår hvad I gerne vil. Her en løsning som jeg tror er i samme ånd, men som vil være nemmere at implementere:

1. Alle tal- og tekstværdier (kolonne C og D) skrives ved udtræk også under "ny værdi" kolonne F.
2. Vi fjerner "Ikke besøgt" kolonnen
3. Ved revision af et punkt ændres værdierne i kolonne F som herved angiver at punktet er blevet genbesøgt
4. Kun de attributter der ændres opdateres i FIRE

Kan det fungere? @busstoptaktik du må også hellere lige vurdere om det er en gangbar vej. Jeg har ikke kigget super grundigt på koden bag ilæg-revision.

> Måske skal "uge" helt udelades fremadrettet, ser ikke den har nogen værdi som sådan.

I kan skrive præcis hvad i vil under ATTR:bemærkning. At der står uge i de fleste nu er bare et levn fra REFGEO. Så gør det endeligt på en smartere måde!

**Reply by ReneTandsbjerg** [reply]
Kan i forklare "Ikke besøgt" kolonnen
Altså hvilken funktion har x? 
Sørger x´et for at tekster ikke bliver læst i forbindelse med ilæg revision eller har den kun relevans for brugeren i marken.

**Reply by kbevers** [reply]
Når der står et x i "ikke besøgt" kolonnen, så fortæller du `fire niv ilæg-revision` at dette punkt ikke er besøgt, og dermed ikke ændret. Altså bliver punkter med et x IKKE skrevet til databasen. Det er den nuværende funktionalitet. Men som beskrevet ovenfor kan vi gøre det på anden vis hvis det er nemmere at arbejde med.

**Reply by ReneTandsbjerg** [reply]
så vil det være bedst at beholde kolonne H og fjerne x, så lade det være brugeren i marken der sætter et x ved de af punkterne som vi ikke ændre ved.

**Reply by kbevers** [reply]
@ReneTandsbjerg vil du ikke lige prøve at forholde dig til den løsning jeg har beskrevet ovenfor?

**Reply by busstoptaktik** [reply]
Idéen med "ikke besøgt" var at i de mange tilfælde hvor kommentaren er "Punkt besøgt uge XX/YYYY, navn-på-revisor", så behøver man ikke skrive noget som helst, bare slette "x" og få kommentaren genereret automatisk.

Mht. "gentagelse" af søjle C/D i søjle F: Det var der vi startede, og det gør det utroligt vanskeligt at se hvor man *har* været: Det er trods alt de færreste elementer der skal ændres fra gang til gang.

Det kan dog sagtens gøres mht. søjle D. Det kniber mere med søjle C: Talværdier er vanskelige at håndtere, fordi den indre repræsentation i computeren ikke er sammenlignelig med det, der står i søjlen (det indre er i 2-talssystemet, som kun er delvist sammenligneligt med 10-talssystemet).

Det er den egentlige grund til at der findes en søjle C og altså C og D ikke bare er fælles. Der findes simple, men farlige løsninger på problemet, og der findes ufarlige, men *helt utroligt* komplicerede løsninger. Hvis vi vidste at alt hvad vi indtastede af tal lå imellem fx 1 og 10, og alt skulle gives med fx 4 decimaler, så var problemet let. Men det er ikke det, der er problemet.

Men prøv lige at overveje dette: Et element fra søjle D kan kopieres til redigering i søjle F ved at trykke ctrl+c, højrepil, højrepil, ctrl+v. Her vil man undgå at der står noget i søjle F som ikke eksplicit er blevet ændret, dvs. bedre overblik over hvad man har rettet og hvad der forbliver som det var.

Både manuel kopiering af det relevante og automatisk kopiering af det hele kan lade sig gøre.

Med automatisk kopiering af det hele slipper man for 4 tastetryk. Til gengæld kan man ikke umiddelbart se hvad man har rettet.

Med manuel kopiering er det omvendt: 4 ekstra tastetryk pr. ændring. Til gengæld er det umiddelbart indlysende hvor der er rettet og hvor der ikke er sket ændringer.

**Comment by busstoptaktik** [comment]
@ReneTandsbjerg - du skriver at du og Palle som **Plan A** gerne vil:

> revisions udtrækket ind i QGIS (liggende lokalt), for der igennem at rette og tilføje i QGIS

Men har du tænkt på at det, der vil ske i QGIS bare er, at for hvert enkelt punkt kan man klikke, og så vil der poppe en attribut*tabel* op som ret nøjagtigt afspejler det, der står i regnearket i dag (det er jo de samme data, der skal revideres på samme måde), men minus muligheden for umiddelbart at overskue hvilke punkter, der er besøgt?

Det kan sagtens lade sig gøre, men det kræver megen kode for en meget lille reel ændring.

**Reply by ReneTandsbjerg** [reply]
Godt så.
Det jeg gerne vil have er at det er en beskrivelse for punktet der popper op når jeg klikker på prikken i QGIS, for så kan jeg nemlig læse hvor punktet sidder og sammen med " hvor er jeg gps prik" i forhold til punkt kan en kombination af de 2 gøre at punktet (kalot, lodret, skruepløk) findes temmelig hurtigt. Vi skal jo snart til at bruge QGIS til punkt-revision i stedet for visfix.

Så hvor om alting er, bliver det så Plan B der køre.? 

1. Alle tal- og tekstværdier (kolonne C og D) skrives ved udtræk også under "ny værdi" kolonne F.
2. Vi beholder kolonne H og fjerne x, og så lade det være brugeren i marken der sætter et x ved de af punkterne som vi ikke ændre ved.
3. Ved revision af et punkt ændres værdierne i kolonne F som herved angiver at punktet er blevet genbesøgt
4. Kun de attributter der ændres opdateres i FIRE

@Palle-N @kbevers

**Reply by busstoptaktik** [reply]
> Det jeg gerne vil have er at det er en beskrivelse for punktet der popper op når jeg klikker på prikken i QGIS, for så kan jeg nemlig læse hvor punktet sidder og sammen med " hvor er jeg gps prik"

Yes, det var præcis intentionen med det overordnede design, så der er lavet plads til at det kan indføres. Som alle andre tilføjelser er det naturligvis et spørgsmål om prioritering af nytteværdi versus kodemængde. Her er kodemængden lille (for tilføjelse af beskrivelse og afmærkning: GPS-prik er ikke på tegnebrættet endnu) og, går jeg ud fra, nytteværdien høj?

**Reply by kbevers** [reply]
> GPS-prik er ikke på tegnebrættet endnu

Den er der vist allerede, det er i hvert fald indbygget (måske via plugin?) funktionalitet i QGIS. Kræver bare man har koblet en GPS-modtager til computeren.

**Reply by ReneTandsbjerg** [reply]
Korrekt @kbevers den funktionalitet er der allerede i QGIS, men som nævnt tidligere ku det være fedt i QGIS at klik på punkt, den valide beskrivelse popper op læse, finde og rette til i (F)Excel-arket og gem.  
Jeg UN hvorvidt det kan gøres med bsk og selvfølgelig offline. Men hvis der er nogle QGIS hajer der ved hvordan skærmkort er tilgængeligt offline og hvorledes beskrivelse knyttes til punktet, tages der imod med åbne arme.  

Emnet begynder at falde lidt udenfor Step 2)
